python/cudf/cudf/core/udf/utils.py

```diff
@@ -213,6 +230,37 @@ def _get_kernel(kernel_string, globals_, sig, func):
     globals_["f_"] = f_
     exec(kernel_string, globals_)
     _kernel = globals_["_kernel"]
-    kernel = cuda.jit(sig)(_kernel)
+    kernel = cuda.jit(sig, link=files, extensions=arg_handlers)(_kernel)
 
     return kernel
+
+
+launch_arg_getters: Dict[Any, Any] = {}
+
+
+def _get_input_args_from_frame(fr):
+    args = []
+    offsets = []
+    for col in _supported_cols_from_frame(fr).values():
+        getter = launch_arg_getters.get(col.dtype)
+        if getter:
+            data = getter(col)
+        else:
+            data = col.data
+        if col.mask is not None:
+            # argument is a tuple of data, mask
+            args.append((data, col.mask))
+        else:
+            # argument is just the data pointer
+            args.append(data)
+        offsets.append(col.offset)
+
+    return args + offsets
+
+
+def _return_col_from_dtype(dt, size):
+    return cp.empty(size, dtype=dt)
+
+
+def _post_process_output_col(col, retty):
+    return as_column(col, retty)
```

Comment:
Do we anticipate these functions containing more extensive logic at some point? If not, I'm not sure that I like their introduction since it's not immediately obvious what post processing or "returning a col from a dtype" should be doing.

Reply:
Yes! This will eventually do the inverse of `to_string_view_array`: a function `from_dstring_array` which is coming when functions that return strings arrive. At that point we'll have

```python
if retty == np.dtype('object'):
    return from_dstring_array(col)
```